tawazi/ops.py

```diff
@@ -196,12 +65,11 @@ def to_dag(declare_dag_function: Callable[..., Any]) -> Callable[..., Any]:
 
     def wrapped(*args: Any, **kwargs: Any) -> DAG:
         # TODO: modify this horrible pattern
-        global exec_nodes
         with exec_nodes_lock:
-            exec_nodes = []
+            node.exec_nodes = []
             declare_dag_function(*args, **kwargs)
-            d = DAG(exec_nodes)
-            exec_nodes = []
+            d = DAG(node.exec_nodes)
```

Comment:
is it because you have circular dependencies that you do `node.exec_nodes` ?

Reply:
I used exec_nodes like this in order to modify `exec_nodes` (it is a variable inside the `node` module inside tawazi) It doesn't seem that I can modify it using any other technique

Reply:
I thought capital Letters are only used for constants... this is not a constant